(maint) Update clj-parent to 1.4.3

Sharpie opened the pull request:
This patch updates the clj-parent dependency to 1.4.1 and drops a now-obsolete
exclusion of javax.servlet/servlet-api from the ring/ring-defaults
dependency.

Sharpie committed:
(maint) Update clj-parent to 1.4.3

This patch updates the clj-parent dependency to 1.4.3 and drops a now-obsolete
exclusion of javax.servlet/servlet-api from the ring/ring-defaults
dependency.

justinstoller commented:
https://github.com/puppetlabs/trapperkeeper-status/pull/84 should fix the error seen in this PR

Magisus commented:
That PR is merged, this will need a rebase.

Sharpie commented:
Hmm. Looks like the test is still failing after rebase, but I can't reproduce locallay using `lein test`. Failure looks related to the status logging added in [TK-400](https://tickets.puppetlabs.com/browse/TK-400)

Sharpie committed:
(maint) Test against Java 8

This patch moves the Travis test environment to Java 8 as Java 7 has reached
end of life.

Sharpie commented:
@justinstoller @Magisus Looks like tests were failing due to Travis still using Java 7 as the test platform. I added another commit to bump the tests to Java 8.

justinstoller commented:
Do we want to update this to tk-status 2.0 since we're dropping Java 7 support?

Sharpie commented:
Looks like we went with a .y bump for the same sort of change in puppetlabs/trapperkeeper-metrics#52. Java 7 support has been functionally dead for a while now given that clj-parent pulls in components that won't work with it --- and that's what the top-level projects have been running with.

justinstoller commented:
I don't think I agree with what we did there. I think Jeremy [originally meant](https://github.com/puppetlabs/trapperkeeper-metrics/pull/52#issuecomment-307971433) for us to bump the Y version but not drop Java 7 support, only drop testing of it, and if it breaks then add it back in a Z. I think we then later came back around and released it, but wrote that we were dropping Java 7 support without thinking much about it.

I would prefer that we followed the approach taken with [tk-jetty9](https://github.com/puppetlabs/trapperkeeper-webserver-jetty9/commit/3128c2b9ebf97dc3ae49ed21574c76caa8b87e50) and [server](https://tickets.puppetlabs.com/browse/SERVER-1765) and bump the X release. Not the approach taken the first week I was back on the team (and when Jeremy and Ruth were on their way out).
